### PR TITLE
Allow Operator Management Commands to work in TLS-Only Environments

### PR DESCRIPTION
**Pull request by jkatz**
**Checklist:**

 
 - [x] Have you added an explanation of what your changes do and why you'd like them to be included?
 - [x] Have you updated or added documentation for the change, as applicable?
 - [x] Have you tested your changes on all related environments with successful results, as applicable?

**Type of Changes:**

 
 - [x] Bug fix (non-breaking change which fixes an issue)
 - [x] New feature (non-breaking change which adds functionality)
 - [x] Breaking change (fix or feature that would cause existing functionality to change)

**What is the current behavior? (link to any open issues here)**

There were a few PostgreSQL Operator operations that made connections over TCP to PostgreSQL clusters, which, if said PostgreSQL clusters were run in "TLS-only" mode (aka `hostssl`), the commands would fail. These included:

- `pgo df`
- `pgo [create|update|delete|show] user`
- `pgo [create|delete] pgbouncer`

**What is the new behavior (if this is a feature change)?**

tl;dr the PostgreSQL Operator now supports clusters whose remote connections support TLS-only :smile:

Numerous changes were made to support this workflow, include:

## Enhanced `pgo df`

`pgo df` provides information on the disk utilization of a PostgreSQL cluster,
and previously, this was not reporting accurate numbers. The new `pgo df` looks
at each PVC that is mounted to each PostgreSQL instance in a cluster, including
the PVCs for tablespaces, and computers the overall utilization. Even better,
the data is returned in a structured format for easy scraping. This
implementation also leverages Golang concurrency to help compute the results
quickly.

## Enhanced pgBouncer Integration

The pgBouncer integration was completely rewritten to support the TLS-only
operations via the PostgreSQL Operator. While most of the work was internal,
you should now see a much more stable pgBouncer experience. Additionally, a few
new commands were added:

- `pgo show pgbouncer` shows information about a pgBouncer deployment
- `pgo update pgbouncer --rotate-password` allows one to rotate the service
account password for pgBouncer

## Rewritten pgo User Management commands

The user management commands were rewritten to support the TLS only workflow.
These commands now return additional information about a user when actions are
taken. Several new flags have been added too, including the option to view all
output in JSON. Other flags include:

- `pgo update user --rotate-password` to automatically rotate the password
- `pgo update user --disable-login` which disables the ability for a PostgreSQL user to login
- `pgo update user --enable-login` which enables the ability for a PostgreSQL user to login
- `pgo update user --valid-always` which sets a password to always be valid, i.e. it has no
expiration
- `pgo show user` does not show system accounts by default now, but can be made
to show the system accounts by using `pgo show user --show-system-accounts`

A major change as well is that the default password expiration function is now
defaulted to be unlimited (i.e. never expires) which aligns with typical
PostgreSQL workflows.

**Other information**:

Issue: [ch7131]
Issue: #1180

### CODE REVIEW

File: ansible/roles/pgo-operator/files/pgo-configs/pgbouncer-template.json [hunk 1]
```diff
@@ -63,9 +55,6 @@
                     }, {
                         "name": "PG_PRIMARY_SERVICE_NAME",
                         "value": "{{.PrimaryServiceName}}"
-                    }, {
-                        "name": "PG_REPLICA_SERVICE_NAME",
```
Comment (cbandy) [inline ansible/roles/pgo-operator/files/pgo-configs/pgbouncer-template.json]: I see this environment variable referenced in [the docs](https://github.com/CrunchyData/postgres-operator/blob/54229ff17e583274025c120380c699e4de8b2e1e/hugo/content/pgo-client/common-tasks.md#connection-pooling-via-pgbouncer); is it safe to remove?

> If you want [PgBouncer] to have an entry for the replica service, add the following configuration to pgbouncer.ini…

File: util/secrets.go [hunk 2]
```diff
@@ -67,16 +84,56 @@ func stringWithCharset(length int, charset string) string {
...
+
 // GeneratePassword generate a password of a given length
 func GeneratePassword(length int) string {
 	return stringWithCharset(length, charset)
 }
 
+// GeneratePostgreSQLMD5Password takes a username and a plaintext password and
+// returns the PostgreSQL formatted MD5 password, which is:
+// "md5" + md5(password+username)
+func GeneratePostgreSQLMD5Password(username, password string) string {
```
[the remark this answers is not in the document]
Comment (jkatz) [reply util/secrets.go]: But it's not encrypted, it's hashed. PostgreSQL used to support encryption via `crypt` but that was removed long ago.

In this case, we are generating the md5 digest that PostgreSQL expects to use as part of its MD5 authentication. So I'm inclined to leave it as is.

File: util/secrets.go [hunk 3]
```diff
@@ -160,26 +217,18 @@ func (cs CloneClusterSecrets) Clone() error {
...
-	err = CreateSecret(clientset, clustername, secretName, username, enPassword, namespace)
-	if err != nil {
-		log.Error("error creating secret" + err.Error())
+
+	if err := CreateSecret(clientset, clustername, secretName, username, password, namespace); err != nil {
+		log.Error(err)
 	}
 
-	return err
+	return nil
```
[the remark this answers is not in the document]
Comment (jkatz) [reply util/secrets.go]: Nope. I added the `err` to the `if` block.

File: events/eventtype.go [hunk 4]
```diff
@@ -395,58 +396,6 @@ func (lvl EventCreateBackupCompletedFormat) String() string {
 	return msg
 }
 
-//----------------------------
-type EventCreateUserFormat struct {
```
[the remark this answers is not in the document]
Comment (jkatz) [reply events/eventtype.go]: Yes.

File: ansible/roles/pgo-operator/files/pgo-configs/pgbouncer-template.json (hunk 1 above)
Comment (jkatz) [reply ansible/roles/pgo-operator/files/pgo-configs/pgbouncer-template.json]: The methodology has shifted now that we have the `*` flag being used to match all available databases.

I went ahead and remove the line from the documentation.
